### ISSUE TITLE
DNDBeyond - Scorching Ray should roll like Magic Missile

### ISSUE THREAD

**Issue by trey-kirk-sp** [mr-ice/maptool-macros#98]
Scorching Ray casts 3 rays, each with an attack. The higherLevelDefinition is empty, so it needs to act like MMissile where additional rays are added when up-cast.

**Comment by trey-kirk-sp** [comment]
Plugging Scorching Ray into the existing logic that captures MMissile doesn't quite do the deed. Where MMissile would create a single damage expression rolled x times was fine, what we really need are x sets of expressions. That way the attack expression gets rolled. 

The not good version:
```
[
    {
    "name": "Scorching Ray",
    "bonus": 10,
    "diceSize": "20",
    "diceRolled": "1",
    "expressionTypes": "Attack",
    "description": "",
    "output": "Rolled: 1 Scorching Ray Attack Roll: 1d20 + 10 = 11 ",
    "rolls": [1],
    "roll": 1,
    "rollString": "1d20 + 10",
    "total": 11,
    "totals": [11],
    "outputs": ["Rolled: 1 Scorching Ray Attack Roll: 1d20 + 10 = 11 "],
    "allTotal": 11
  },
    {
    "name": "Scorching Ray",
    "diceSize": 6,
    "diceRolled": 2,
    "expressionTypes": "Damage",
    "bonus": "0",
    "totalRolls": 8,
    "damageTypes": "Fire",
    "output": "Rolled: 9 Scorching Ray Damage Roll: 2d6 = 9 Fire",
    "rolls":     [
      6,
      5,
      8,
      5,
      4,
      6,
      6,
      9
    ],
    "roll": 9,
    "rollString": "2d6",
    "total": 9,
    "totals":     [
      6,
      5,
      8,
      5,
      4,
      6,
      6,
      9
    ],
    "outputs":     [
      "Rolled: 6 Scorching Ray Damage Roll: 2d6 = 6 Fire",
      "Rolled: 5 Scorching Ray Damage Roll: 2d6 = 5 Fire",
      "Rolled: 8 Scorching Ray Damage Roll: 2d6 = 8 Fire",
      "Rolled: 5 Scorching Ray Damage Roll: 2d6 = 5 Fire",
      "Rolled: 4 Scorching Ray Damage Roll: 2d6 = 4 Fire",
      "Rolled: 6 Scorching Ray Damage Roll: 2d6 = 6 Fire",
      "Rolled: 6 Scorching Ray Damage Roll: 2d6 = 6 Fire",
      "Rolled: 9 Scorching Ray Damage Roll: 2d6 = 9 Fire"
    ],
    "allTotal": 49
  }
]
```

**Comment by trey-kirk-sp** [comment]
Addressed it by using the same logic to determine totalRolls and instead applying that to creating new expressions. So where Magic Missile was one Damage expression with a totalRoll set to 3, it's now an array of 3 Damage expressions (rolling 1, each). This applied to Scorching Ray by creating an array of Attack/Damage pairs of expressions. It works, but the output looks super clunky.

![image](https://user-images.githubusercontent.com/6287933/88127340-fb4fcd00-cb98-11ea-9634-973d52d02d1d.png)

Gonna call this "fixed" and address the output either with Jay fixing it or stealing his new fangled Attack output code when he checks it in.

**Comment by trey-kirk-sp** [comment]
Fixed in hash e10badc4
